File: java-codellama7B-FP-all/taskid-6_parse_nested_parens-gen14.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    // Input to this function is a string represented multiple groups for nested parentheses separated by spaces.
    // For each of the group, output the deepest level of nesting of parentheses.
    // E.g. (()()) has maximum two levels of nesting while ((())) has three.
    // >>> parseNestedParens(("(()()) ((())) () ((())()())"))
    // (new ArrayList<Long>(Arrays.asList((long)2l, (long)3l, (long)1l, (long)3l)))
    public static ArrayList<Long> parseNestedParens(String paren_string) {

        char[] parens = paren_string.toCharArray();
        ArrayList<Long> levels = new ArrayList<>();
        int level = 0;
        for (int i = 0; i < parens.length; i++) {
            if (parens[i] == '(') {
                level++;
                if (level > levels.size()) {
                    levels.add(0l);
                }
                levels.set(level-1, levels.get(level-1) + 1l);
            } else if (parens[i] == ')') {
                level--;
                if (level >= 0) {
                    levels.set(level, levels.get(level) - 1l);
                }
            }
        }
        return levels;
    }
    public static void main(String[] args) {
    assert(parseNestedParens(("(()()) ((())) () ((())()())")).equals((new ArrayList<Long>(Arrays.asList((long)2l, (long)3l, (long)1l, (long)3l)))));
    assert(parseNestedParens(("() (()) ((())) (((())))")).equals((new ArrayList<Long>(Arrays.asList((long)1l, (long)2l, (long)3l, (long)4l)))));
    assert(parseNestedParens(("(()(())((())))")).equals((new ArrayList<Long>(Arrays.asList((long)4l)))));
    }

}

File: py-codellama13B-FP-all/taskid-6_parse_nested_parens-gen14.py
from typing import List

def parse_nested_parens(paren_string: str) -> List[int]:
    """ Input to this function is a string represented multiple groups for nested parentheses separated by spaces.
    For each of the group, output the deepest level of nesting of parentheses.
    E.g. (()()) has maximum two levels of nesting while ((())) has three.

    >>> parse_nested_parens('(()()) ((())) () ((())()())')
    [2, 3, 1, 3]
    """

    max_nesting_level = []
    for i in paren_string.split():
        current_nesting_level = 0
        for j in i:
            if j == '(':
                current_nesting_level += 1
            if j == ')':
                current_nesting_level -= 1
        max_nesting_level.append(current_nesting_level)
    return max_nesting_level
